### PR TITLE
Test on Python 3.7 & 3.8 and speed up Travis

### PR DESCRIPTION
**Pull request by imjoehaines**
## Goal

Due to changes in Travis' build images and Python package updates, the Python 2.6 and 3.3 jobs on Travis don't work in master

This PR fixes them by using `dist: trusty` so they are installable again and pinning dependencies on 3.3

I've also collapsed all of the jobs into one job per Python version, which speeds up the test suite from taking [~35 minues](https://travis-ci.com/github/bugsnag/bugsnag-python/builds/163724474) to taking [~13 minutes](https://travis-ci.com/github/bugsnag/bugsnag-python/builds/163732556). This makes the Travis file a bit gross, but AFAICT that's the only downside — tox still runs each in a separate environment and coveralls reports the same coverage

### Changed

- Travis is now running one build per Python version
- Python 3.7 and 3.8 are running on Travis
- tox no longer lists environments which aren't valid

## Review

For the submitter, initial self-review:

- [x] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: tests/test_delivery.py [hunk 1]
```diff
@@ -36,40 +37,6 @@ def test_urllib_delivery(self):
...
-            raise SkipTest("Requests is not installed")
-
-        self.config.configure(endpoint=self.server.url)
-        del self.config.use_ssl
-        RequestsDelivery().deliver(self.config, '{"legit": 4}')
-
-        self.assertSentReportCount(1)
-        request = self.server.received[0]
-        self.assertEqual(request['json_body'], {'legit': 4})
-
```
Comment (imjoehaines) [inline tests/test_delivery.py:72]: These deleted tests now live in `tests/requests/test_requests_delivery.py`

File: tests/test_notify.py [hunk 2]
```diff
@@ -168,15 +168,16 @@ def test_notify_user_filter(self):
         self.assertEqual('Man', event['user']['lastname'])
 
     def test_notify_payload_matching_filter(self):
-        bugsnag.configure(params_filters=['number'])
-        bugsnag.notify(ScaryException('unexpected failover'),
-                       apple='four', number=76)
+        bugsnag.configure(params_filters=['bar'])
+        bugsnag.notify(ScaryException('unexpected failover'), foo=4, bar=76)
```
Comment (imjoehaines) [inline tests/test_notify.py:172]: This change is to pull all of the `bugsnag.notify` call onto one line, which gets around a small difference in how Python 3.8 reports line numbers in stack traces — see b4f3095664456572af1b83da691bded0f13222fb and https://bugs.python.org/issue12458 for more information

I think this is a bit nicer than doing a dynamic version check to see if the line number should be 172 or 173

File: tox.ini [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
+    py{26,27,33,34,35,36,37,38}-requests,
     py{26,27}-flask,
-    py{26,27,33,34,35,36}-wsgi,
-    py{27,36}-django{18,19,110,111,20,21}-migrate1-django1
-    py36-lint,
+    py{26,27,33,34,35,36,37,38}-wsgi,
+    py{35,36,37}-django{18,19,110,111,20,21}-migrate1-django1,
+    py27-django{18,19,110,111}-migrate1-django1,
+    py38-django{18,111,20,21}-migrate1-django1,
+    py38-lint
```
Comment (imjoehaines) [inline tox.ini:10]: I've split out these to remove combinations that aren't valid, e.g. `py27-django20-migrate1-django1` doesn't work so it's not longer here
